### PR TITLE
 Support DeclType and AttrExportType

### LINKED ISSUE #12: Absform: -export_type attribute

### LINKED ISSUE #13: Absform: -type attribute

### LINKED ISSUE #14: Absform: -opaque attribute

### PR DESCRIPTION
**Pull request by youxkei**
This PR resolves #12, resolves #13, and resolves #14.

### CODE REVIEW

File: lib/abstract_format.ml [hunk 1]
```diff
@@ -23,13 +23,14 @@ and form_t =
...
   | AttrExport of line_t * (string * int) list
+  | AttrExportType of line_t * (string * int) list
   | AttrImport of line_t * (string * int) list
   | AttrMod of line_t * string
   | AttrFile of line_t * string * line_t
   | DeclFun of line_t * string * int * clause_t list
   | SpecFun of line_t * string option * string * int * type_t list
   | DeclRecord of line_t * (line_t * string * expr_t option * type_t option) list
-  | DeclType
+  | DeclType of line_t * bool * string * (line_t * string) list * type_t
```
Comment (yutopp) [inline lib/abstract_format.ml]: NOTE: A `bool` parameter at the 2nd position becomes `true` when the declaration is a `opaque`, right?

File: lib/abstract_format.ml [hunk 2]
```diff
@@ -182,20 +191,31 @@ and form_of_sf sf =
...
+
   (* eof *)
   | Sf.Tuple (2, [Sf.Atom "eof"; Sf.Integer line]) ->
      FormEof
 
   | _ ->
      raise_unknown_error "form" sf
 
-and fa_list_of_sf sf =
+and name_and_arity_of_sf sf =
```
Comment (yutopp) [inline lib/abstract_format.ml]: :+1:

File: lib/abstract_format.ml (hunk 1 above)
Comment (youxkei) [reply lib/abstract_format.ml]: Yes, you're right.
The second parameter means something like `is_opaque`.

Comment (yutopp) [reply lib/abstract_format.ml]: If so, would you add a comment about  a meaning of that or use a data type (e.g. `AttrType | AttrOpacue`) to clarify the meaning, please? Do you have any ideas?

**Comment by yutopp** [comment]
LGTM with nits! Thank you!

File: lib/abstract_format.ml [hunk 3]
```diff
@@ -174,28 +184,50 @@ and form_of_sf sf =
...
+                                t;
+                                Sf.List tvars
+                               ]);
+                 ]) ->
+    DeclType (line, name, tvars |> List.map tvar_of_sf, type_of_sf t)
+
+  (* opaque type declaration *)
+  | Sf.Tuple (4, [Sf.Atom "attribute";
+                  Sf.Integer line;
+                  Sf.Atom attribute;
```
Comment (yutopp) [inline lib/abstract_format.ml]: `attribute` -> `”opaque”`?

Comment (youxkei) [reply lib/abstract_format.ml]: I fixed it.

**Comment by yutopp** [comment]
LGTM

**Pull request merged**